### DISCUSSION TITLE
RAM - Density calculations

### DISCUSSION THREAD

**Discussion by consithe1** [Boavizta/boaviztapi#204] [Q&A] [answered]
Hi! It's me again,
As I was looking for RAM die sizes on Internet, I came by [this article](https://www.techinsights.com/blog/memory/micron-1a-dram-technology#:~:text=D1%CE%B1%20DRAM%20Device%20Analysis%20%26%20Competitive,DRAM%20density%20up%20to%20date.).

Unfortunately, the density values in the documentation have two things that I am not sure about:

1. The unit given by the API and in the documentation are different (the API call writes it as "GB/cm2" while the documentation says "cm2/Go". 
2. If I take for example the density and capacity of a [8GB DDR4 RAM](https://www.techinsights.com/blog/memory/micron-1a-dram-technology#:~:text=D1%CE%B1%20DRAM%20Device%20Analysis%20%26%20Competitive,DRAM%20density%20up%20to%20date.), then `capacity = 8 GB` and `density = 0.299 GB/mm²`. Which then leads to find that the `die_size` should be around 26.76 mm². This value seems coherent since some die size values in the article seem close to it. However, converted in GB/cm² (which is the unit used by the API), this value should be close to `29.9 GB/cm²` which is far from the default value of `0.625 GB/cm²`. 
3. How does the size of the processes in the RAM have influence on the manufacture impact please?
4. Based on the formula of the [RAM manufacture impact](https://github.com/Boavizta/boaviztapi/blob/main/docs/docs/Explanations/components/ram.md#manufacture-impact), by doing a dimensional analysis, I find this: 
```math
[kgCO2eq] \ne \frac{[GB]}{[\frac{GB}{cm²}]} * [kgCO2eq] + [kgCO2eq] \rightarrow [kgCO2eq] \ne [cm²] * [kgCO2eq] + [kgCO2eq]
```
Which means that I should still have `cm²` in the final result which seems incoherent

Thank you in advance for your reply,
Best regards

**Comment by da-ekchajzer** [comment]
1. It's a typo in the documentation, thank you very much for your feedback. And we should use an international unit (GB instead of Go). Corrected in :  1022a38

2. I'll try to have a look this week and give you a feedback

3. Today in the API the ```process``` is a way to filter on the available existing densities in the following file : https://github.com/Boavizta/boaviztapi/blob/main/boaviztapi/data/components/ram_manufacture.csv when density is not given.

In a proprietary LCA database, you would benefit from impact factors that are specific to each process. In such cases, the process would be helpful to select the right impact factor. In our case (open source data) we don't have such a level of precision. We use an open impact factor that take into account a 20 nm process.

4. The impact factor ram_manufacture_die is in kgCO2eq/cm2. So we and up by having kgCO2eq: 

```math
\frac{[GB]}{[\frac{GB}{cm²}]} * \frac{[kgCO2eq]}{cm²} + [kgCO2eq] \rightarrow [cm²] * \frac{[kgCO2eq]}{cm²} + [kgCO2eq] 
```

**Reply by da-ekchajzer** [reply, marked as answer]
> If I take for example the density and capacity of a [8GB DDR4 RAM](https://www.techinsights.com/blog/memory/micron-1a-dram-technology#:~:text=D1%CE%B1%20DRAM%20Device%20Analysis%20%26%20Competitive,DRAM%20density%20up%20to%20date.), then capacity = 8 GB and density = 0.299 GB/mm². Which then leads to find that the die_size should be around 26.76 mm². This value seems coherent since some die size values in the article seem close to it. However, converted in GB/cm² (which is the unit used by the API), this value should be close to 29.9 GB/cm² which is far from the default value of 0.625 GB/cm².

I think you're falling for the classic byte (GB) vs. bit (Gb) fallacy 😉.

In your article, the density is given in bit (Gb/mm2) and not Byte (GB/mm2). To convert Gb/mm2 to GB/cm2 you have to multiply by 100 and divide by 8. It means that 1Gb/mm2 = 12.5 GB/cm². So the density given is 0.299 Gb/mm² * 12.5 = 3.7375 GB/cm2. 

You might say that it is still not the same order of magnitude as the density we use. I believe that a density of 0.299 is a very recent density. If you look at the density in the following table, you see that pre-2017 densities are not bigger than 1.9 Gb/mm².

![image](https://github.com/Boavizta/boaviztapi/assets/24867893/59d3660b-284b-4161-8345-a889d9bbc32b)

Hope I am right and clear. Please let me know if not.

We would be happy to have you on the public chat to know a little more about your use case since it seems you have been digging in the API the last couple of days.

**Reply by consithe1** [reply]
Hello!
Thank you so much for taking time to reply (especially when I make such mistakes).

1. 
2. I am honestly ashamed of having confused the two of them, but thank you very much for pointing it out!
3. This makes a lot of sense, thank you!
4. And once again, sorry for such mistake. I must have considered that the `ram_manufacture_die` was of the same unit that the `ram_manufacture_base`

I won't forget to tell you about my use case in the public chat in the following days!
Best regards,

**Comment by da-ekchajzer** [comment]
Nothing to be ashamed, we all did the Byte vs. bit confusion at some point. Happy that it is getting clearer for you.
